spring-kafka/src/main/java/org/springframework/kafka/core/KafkaOperations.java

```diff
@@ -64,6 +64,16 @@
 	ListenableFuture<SendResult<K, V>> sendDefault(int partition, K key, V data);
 
 	/**
+	 * Send the data to the default topic with the provided key and partition.
+	 * @param partition the partition.
+	 * @param timestamp the timestamp of the record.
+	 * @param key the key.
+	 * @param data the data.
+	 * @return a Future for the {@link SendResult}.
+	 */
+	ListenableFuture<SendResult<K, V>> sendDefault(int partition, Long timestamp, K key, V data);
```

Comment:
Yes, makes sense @artembilan, will make partition also an `Integer`. I see one problem because of this change though, there is ambiguity in these api's(if the key is an Integer type):

`ListenableFuture<SendResult<K, V>> send(String topic, Integer partition, V data);`
`ListenableFuture<SendResult<K, V>> send(String topic, K key, V data);`, 

How about eliminating `ListenableFuture<SendResult<K, V>> send(String topic, Integer partition, V data);` altogether and keeping `ListenableFuture<SendResult<K, V>> send(String topic, Integer partition, K key, V data);` - this maps with ProducerRecord constructors too